chemreg/jsonapi/views.py

```diff
@@ -34,7 +34,15 @@ def create(self, request, *args, **kwargs):
 class ModelViewSet(
     ValidatePostParamsMixin, FormatRelatedFieldMixin, views.ModelViewSet
 ):
-    pass
+    def get_permissions(self):
+        # return permission_classes depending on `action`
+        if "cid" in self.request.data:
+            return [
+                permission()
+                for permission in self.permission_classes_by_action[self.action]
+            ]
+        else:
+            return [permission() for permission in self.permission_classes]
```

Comment:
It took me a second to find this. This will end up getting applied to every view in the project going forward. Could we apply it to only the `DefinedCompoundViewSet` and `IllDefinedCompoundViewSet`? Maybe via a mixin?

Reply:
that makes sense, I'll move it into `/compound/models.py`